NotificationServiceExtension/MultisigNotification+ContentProvider.swift

```diff
@@ -0,0 +1,197 @@
+//
+//  MultisigNotification+ContentProvider.swift
+//  Multisig
+//
+//  Created by Dmitry Bespalov on 26.07.21.
+//  Copyright © 2021 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import CoreData
+import SwiftCryptoTokenFormatter
+
+protocol NotificationContentProvider {
+    func loadContent(completion: @escaping ((title: String, body: String)?) -> Void)
+}
+
+extension MultisigNotification: NotificationContentProvider {
+    func loadContent(completion: @escaping ((title: String, body: String)?) -> Void) {
+        switch self {
+        case let .incomingNativeCoin(n):
+            n.loadContent(completion: completion)
+        case let .incomingToken(n):
+            n.loadContent(completion: completion)
+        case let .executedMultisigTransaction(n):
+            n.loadContent(completion: completion)
+        case let .newConfirmation(n):
+            n.loadContent(completion: completion)
+        case let .confirmationRequest(n):
+            n.loadContent(completion: completion)
+        case .unknown:
+            completion(nil)
+        }
+    }
+}
+
+extension SafeNotification {
+    func loadSafe(completion: @escaping (Safe?, NSManagedObjectContext) -> Void) {
+        NotificationService.coreData.persistentContainer.performBackgroundTask { context in
+            completion(safe(context: context, address: address, chainId: chainId), context)
+        }
+    }
+
+    func safe(context: NSManagedObjectContext, address: AddressString, chainId: UInt256String) -> Safe? {
+        do {
+            let fr = Safe.fetchRequest().by(address: address.description, chainId: chainId.description)
```

Comment:
Why didn't you use `Safe.by(address: address.description, chainId: chainId.description)` here?

Reply:
because I only extracted minimal functionality in order to keep code dependencies low. Also, I need to use a background context and not a view context

Reply:
👍🏻